server/plugins/LFUGens.cpp

```diff
@@ -2379,33 +2379,126 @@ void EnvGen_Ctor(EnvGen* unit) {
     EnvGen_next_k(unit, 1);
 }
 
+static bool EnvGen_initSegment(EnvGen* unit, int& counter, double level, float overrideDur = -1) {
```

Comment:
document the meaning of these parameters please, also change `float overrideDur` to `double dur` as in the comment below

Reply:
This function was a goto in nextSegment. I changed it to a function to be able to call it also in check_gate and use a supplied dur instead of the one defined by the segment. What do you think? Shall I add a comment explaining that unit, counter and level are from nextSegment and check_gate, while dur allows dur overriding?

Reply:
mainly, i would like to see the behavior of `dur` documented since that behavior, while expedient, is not immediately obvious. explaining the purpose of the other parameters would be helpful too.

Reply:
I put a comment there, please see if it's ok